docs/HowToCertificatesRotation.md

```diff
@@ -0,0 +1,486 @@
+
+Upgrade the Helm release installed with the [database](../stable/database) chart using the new TLS secrets.
+This task will perform a rolling upgrade on all database Storage Manager (SM) statefulsets and Transaction Engine (TE) deployments, therefore ensure that enough additional database processes are running to prevent downtime.
+
+```bash
+helm upgrade database nuodb/database \
+  --namespace nuodb \
+  --set admin.tlsKeyStore.secret=nuodb-keystore-renewed \
+  --set admin.tlsClientPEM.secret=nuodb-client-pem-renewed \
+  --set admin.tlsCACert.secret=nuodb-ca-cert-renewed \
+  -f values.yaml
+```
+
+> **NOTE**: Adjust the values depending on the key pair certificates that have been rotated.
+
+Wait for all database processes to restart, report `Ready` and ensure that the new key pair certificates are used by all domain processes as described in [Verify Domain Certificates](#verify-domain-certificates).
+
+#### Using HashiCorp Vault
+
+HashiCorp uses sidecar containers to inject keys into another container.
+It automatically updates secret values stored in the Vault KV store to match the files in the volume mounted inside the container.
+NuoDB APs will reload the keystore without the need for reconfiguration and restart.
+Therefore, it is enough to update the keystore, client PEM and CA certificate in HashiCorp Vault.
+
+Ensure that the keystore has the same password (as specified by `admin.tlsKeyStore.password` Helm option) so that the APs can reload the keystore without having to be reconfigured and restarted.
+
+For more information on how to configure TLS with HashiCorp Vault, see [Using HashiCorp Vault for Management of TLS Certificates](./HowToHashiCorpVault.md).
+
+Make sure that the new key pair certificates are used by all APs as described in [Verify Domain Certificates](#verify-domain-certificates).
+
+Restart database processes so that the APs generate new engine certificates signed with the new server certificate chain in the [Intermediate CA](./HowtoTLS.md#intermediate-ca) model or the new shared key pair certificates are used in the [Pass-down](./HowtoTLS.md#pass-down) model.
+Kubernetes 1.15+ supports a rollout restart which can be used to restart database processes in a rolling fashion.
+
+```bash
+kubectl rollout restart \
+  --namespace nuodb \
+  statefulset sm-database-nuodb-cluster0-demo-hotcopy
+
+kubectl rollout restart \
+  --namespace nuodb \
+  statefulset sm-database-nuodb-cluster0-demo
+
+kubectl rollout restart \
+  --namespace nuodb \
+  deployment te-database-nuodb-cluster0-demo
+```
+
+Alternatively, the database processes can be restarted manually by shutting down each process using `nuocmd shutdown process --start-id ...` command.
+Kubernetes will automatically restart the engine container which will effectively restart the database process.
```

Comment:
given that this doc is k8s specific, it might be a better idea to suggest deleting pods, rather than using `nuocmd`.

Reply:
Okay